src/macfc-security-hub-sync.ts

```diff
@@ -451,30 +656,58 @@ export class SecurityHubJiraSync {
       summary: newIssueData.fields.summary
     }
   }
+  shouldCreateIssue(finding: SecurityHubFinding, jiraIssues: Issue[]) {
+    const potentialDuplicates = jiraIssues.filter(issue => {
+      if (!finding.title) {
+        return false
+      }
+      const title = finding.title
+      return issue.fields.description?.includes(title)
+    })
+    console.log('Potential Duplicates: ', potentialDuplicates.length)
+    if (potentialDuplicates.length == 0) {
+      return true
+    }
 
+    const final = potentialDuplicates.filter(issue => {
+      const duplicate = finding.Resources?.reduce(
+        (should: boolean, resource: Resource): boolean => {
+          const id = resource.Id ?? ''
+          if (!id) {
+            return false
+          }
+          return should && issue.fields.description?.includes(id) == true
+        },
+        true
+      )
+      return !duplicate
+    })
+
+    return final.length >= 1
+  }
   async createJiraIssuesForNewFindings(
     jiraIssues: Issue[],
     shFindings: SecurityHubFinding[],
     identifyingLabels: string[]
   ) {
     const updatesForReturn: UpdateForReturn[] = []
-    const existingJiraIssueTitles = jiraIssues.map(i => i.fields.summary)
     const uniqueSecurityHubFindings = [
       ...new Set(shFindings.map(finding => JSON.stringify(finding)))
     ].map(finding => JSON.parse(finding))
 
     for (let i = 0; i < uniqueSecurityHubFindings.length; i++) {
       const finding = uniqueSecurityHubFindings[i]
-      if (
-        !existingJiraIssueTitles.includes(
-          `SecurityHub Finding - ${finding.title}`
-        )
-      ) {
-        const update = await this.createJiraIssueFromFinding(
-          finding,
-          identifyingLabels
-        )
-        updatesForReturn.push(update)
+      if (this.shouldCreateIssue(finding, jiraIssues)) {
+        try {
+          const update = await this.createJiraIssueFromFinding(
+            finding,
+            identifyingLabels
+          )
+          updatesForReturn.push(update)
+        } catch (e) {
+          console.log(e)
+          console.log('Moving forward with next findings')
```

Comment:
Is it possible to consolidate the two console.log statements above?

Reply:
Done!